docs/source/howto/codes/interfacing.rst

```diff
@@ -0,0 +1,282 @@
+        codeinfo.code_uuid = input_code.uuid
+        codeinfo.stdout_name = output_filename
+        codeinfo.cmdline_params = ['-in', input_filename]
+
+        calcinfo = CalcInfo()
+        calcinfo.codes_info = [codeinfo]
+        calcinfo.retrieve_list = [output_filename]
+
+        return calcinfo
+
+The second block corresponds to our previously described first step (preparing the inputs to run): here we are using the ``folder.open`` method to get a handle to the file and write in an input file (in the format in which the code will read it) the values stored in our input nodes.
+This folder (along with all the files created in it) will not only be copied to the remote machine for the code to be ran there, but will also be stored in the local repository of the calculation node.
+
+The last two blocks correspond to the second step (instructions for the engine).
+In the first one we are creating the ``codeinfo = CodeInfo()`` object, that will be then part of the ``calcinfo = CalcInfo()`` object, as one can see in ``calcinfo.codes_info = [codeinfo]``.
+The ``code_uuid`` is necessary for the engine to get the base information set up in the |Code| node (in which machine the code is placed, what is the location of the executable, etc).
+Note that this was taken from ``input_code = self.inputs['code']``, which we never specified in the ``define`` method: this is one of the two mandatory inputs defined in the base |CalcJob| that will need to be always provided in the instance.
+Moreover, in the case of the ``inputs['code']``, the designer of the |CalcJob| sub-class will always have to take care to pass its uuid into ``codeinfo.code_uuid`` as shown here.
+
+Of the last 3 lines remaining, the first two are providing part of the instructions to the engine on how to build the running script:
+
+.. code:: python
+
+        codeinfo.stdout_name = output_filename
+        codeinfo.cmdline_params = ['-in', input_filename]
+
+The first one establishes the command line arguments to be passed to the code: in this case this is the only input of our ``add.sh`` script, but more complex codes can receive multiple inputs or flags through here.
+The second one indicates to the engine where to redirect the standard output.
+The combination of the two result in the following line in the running script:
+
+.. code:: bash
+
+    #!/bin/bash
+    # Scheduler commands
+
+    # Pre-running setups
+
+    '${executable_from_code_uuid}' '-in' '${input_filename}' > ${output_filename}
+
+    # Post-running commands
+
+Which will default to ``'add.sh' '-in' 'aiida.in' > 'aiida.out'``.
+Through the |CodeInfo| instanse it is also posible to configure what to pass through the standard input, as well as what commands to run before and after the execution line.
+
+The final line remaining (``calcinfo.retrieve_list = [output_filename]``) just indicates to the engine which files should be retrieved (either for immediate parsing or local storage for future post-processing).
```

Comment:
Sphinx allows you to turn on line numbers for code segments. I'd suggest you turn those on an then reference the lines by number.

Reply:
Noise.